File: 240203/그 요일은/the-day-of-the-day.java
import java.io.*;

import java.util.*;

public class Main {
    static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    static BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
    static StringTokenizer st;
    static StringBuilder sb = new StringBuilder();

    static String dayOfWeek[] = {"Mon", "Tue", "Wed", "Thu", "Fri", "Sat", "Sun"};
    static int dayOfMonth[] = {0, 31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31};

    public static void main(String[] args) throws IOException {

        st = new StringTokenizer(br.readLine());

        int m1 = Integer.parseInt(st.nextToken());

        int d1 = Integer.parseInt(st.nextToken());

        int m2 = Integer.parseInt(st.nextToken());

        int d2 = Integer.parseInt(st.nextToken());

        String day = br.readLine();

        int totalDay = cal(m1, m2, d1, d2);

        int cal = totalDay / 7;

        int dayNum = 0;

        int res = 0;

        for (int i = 0; i < 7; i++) {
            if (dayOfWeek[i].equals(day)) {
                dayNum = i;
            }
        }


        if (totalDay >= dayNum) {
          res += 1;
        }

        res += cal;

        sb.append(res);


        bw.write(sb.toString());
        bw.flush();
        bw.close();


    }

    private static int cal(int m1, int m2, int d1, int d2) {

        int total1 = 0;
        for (int i = 1; i < m1; i++) {
            total1 += dayOfMonth[i];
        }

        total1 += d1;


        int total2 = 0;

        for (int i = 1; i < m2; i++) {
            total2 += dayOfMonth[i];
        }
        total2 += d2;

        int total = total2 - total1;

        if (total < 0) {
            while (total < 0) {
                total += 7;
            }
        }


        return total;

    }


}

File: 250928/그 요일은/the-day-of-the-day.py
# 24년 m1월 d1일이 월요일.
# m2 d2 까지 A요일이 몇번?

m1, d1, m2, d2 = tuple(map(int, input().split()))
day = input()

#월요일부터 목표 요일까지 며칠차이인지 구한다.
#현재 날짜로부터 + 해준다.
# 그날짜부터 목표 날짜까지 며칠걸리는지 구한다.


days = [0, 31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31]

day_of_weeks = ["Mon", "Tue", "Wed", "Thu", "Fri", "Sat", "Sun"]

day_diff = day_of_weeks.index(day)

d1 += day_diff

def get_total_days(m, d) :

    total_days = 0;

    for i in range (1, m) :
        total_days += days[i]

    return total_days + d

total_days = get_total_days(m2,d2) - get_total_days(m1,d1)

calulated_days = total_days // 7 + 1

print(calulated_days)

